tests/cases/compiler/conditionalTypeDiscriminatingLargeUnionRegularTypeFetchingSpeedReasonable.ts

```diff
@@ -0,0 +1,8012 @@
+type BigUnion =
+    {
+        name: '0';

```

Comment:
how long does does the original 1,000 take without the fix? 30 seconds post-fix is still pretty long to wait for a test to run.

Reply:
It's actually timing out on CI because it's taking more than 40 seconds there. I may need to either trim it a bit, or try to see what else is doing poorly and improve the perf more. 40 seconds for a single file with 6 statements and no control flow is _still_ unreasonably bad, just not _as_ unreasonable.